packages/esm-patient-registration-app/src/widgets/display-photo.component.tsx

```diff
@@ -1,26 +1,28 @@
-import React, { useEffect, useState } from 'react';
-import placeholder from '../assets/placeholder.svg';
-import { useConfig } from '@openmrs/esm-framework';
-import { fetchPatientPhotoUrl } from '../patient-registration/patient-registration.resource';
+import React from 'react';
+import Avatar from 'react-avatar';
+import GeoPattern from 'geopattern';
+import { usePatientPhoto } from '../patient-registration/patient-registration.resource';
 import styles from './display-photo.scss';
 
-export default function DisplayPatientPhoto(props: { patientUuid: string }) {
-  const [photo, setPhoto] = useState(placeholder);
-  const config = useConfig();
+interface DisplayPatientPhotoProps {
+  patientName: string;
```

Comment:
Given that we're already fetching the patient photo data using the `patientUuid`, can't we just get the name from the same SWR-cached fetch data, and avoid adding it to the extension props?

Reply:
It's entirely possible - the patient's person object includes the `display` property which contains the primary identifier concatenated with the patient name. Remember that we're rendering the patient's name as a fallback. This strategy is contingent on having the patient name available at the time the Avatar component gets rendered. Relying on a network request would mean having to wait for the request to resolve and needing to show some loading state in the meantime. I don't know if that'd be an ideal experience.

Reply:
But don't we wait for the request to resolve anyway, to find out whether there's a patient picture?

Reply:
The real problem is the endpoint we're using to fetch the patient photo doesn't return anything if there's no observation data for that particular patient. It only resolves the patient's name when there's a picture stored on the server.

Reply:
Okay, thanks for checking @denniskigen